src/service/tcp.rs

```diff
@@ -75,15 +76,27 @@ where
         let req_hdr = req_adu.hdr;
 
         self.framed.send(req_adu).await?;
-        let res_adu = self
-            .framed
-            .next()
-            .await
-            .ok_or_else(Error::last_os_error)??;
+        
+        let mut retries = 0;
+        loop {
+            let res_adu = self
+                .framed
+                .next()
+                .await
+                .ok_or_else(Error::last_os_error)??;
 
-        match res_adu.pdu {
-            ResponsePdu(Ok(res)) => verify_response_header(req_hdr, res_adu.hdr).and(Ok(res)),
-            ResponsePdu(Err(err)) => Err(Error::new(ErrorKind::Other, err)),
+            match res_adu.pdu {
+                ResponsePdu(Ok(res)) => if let Err(e) = verify_response_header(req_hdr, res_adu.hdr) {
+                    if retries >= MAX_RETRIES {
+                        return Err(e);
+                    }
+                    retries += 1;
```

Comment:
The error should at least be logged instead of silently ignoring it.

Reply:
```suggestion
                    if retries >= MAX_RETRIES {
                        return Err(e);
                    }
                    retries += 1;
                    log::info!("Try to recover from header mismatch: expected/request = {:?}, actual/response = {:?}, retries = {:?}", req_hdr, res_adu.hdr, retries);
```

log like this is acceptable?

Reply:
Please modify the code directly instead of replying to comments. Commits can be squashed later before merging.